overlord/hookstate/hookmgr.go

```diff
@@ -143,14 +143,37 @@ func (m *HookManager) Stop() {
 	m.runner.Stop()
 }
 
+func (m *HookManager) snapContext(contextID string) (context *Context, err error) {
+	var contexts map[string]string
+	err = m.state.Get("snap-contexts", &contexts)
+	if err != nil {
+		if err != state.ErrNoState {
+			return nil, fmt.Errorf("failed to get snap contexts: %v", err)
+		}
+	}
+	if snapName, ok := contexts[contextID]; ok {
+		// create new ephemeral context
+		context, err = NewContext(nil, m.state, &HookSetup{Snap: snapName}, nil, contextID)
+		return context, err
+	}
+	return nil, fmt.Errorf("unknown snap context requested")
```

Comment:
This would become something like "invalid snap cookie provided", I think, which is also nicer.